### ISSUE TITLE
IE11: Read/Update single variable

### ISSUE THREAD

**Issue by bene-starzengruber** [jhildenbiddle/css-vars-ponyfill#33]
Hey!
I'm searching for a way on how to read/edit a single css-variable with JS.
I can't just reuse the cssVars function, as this would overwrite my other variables:
` cssVars({
      variables: {
        '--backgroundColor': value
      }
    });`

Is there a way to make single updates or getting all current variables?
Thanks in advance :)

**Comment by jhildenbiddle** [comment]
Hey @bene-starzengruber.

Use the [`options.onComplete`](https://github.com/jhildenbiddle/css-vars-ponyfill#optionsoncomplete) callback to access a list of all CSS custom properties.

```javascript
cssVars({
  onlyLegacy: false, // Execute in modern and legacy browsers
  updateDOM : false, // Don't do anything
  onComplete: function(cssText, styleNode, cssVariables) {
    // Do stuff with cssVariables
    console.log(cssVariables);
  }
});
```

Use [`options.variables`](https://github.com/jhildenbiddle/css-vars-ponyfill#optionsvariables) to add or modify CSS custom properties. See the [Examples](https://github.com/jhildenbiddle/css-vars-ponyfill#examples) section for details.

```javascript
cssVars({
  variables: {
    "--my-color": "red"
  }
});
```

This will either add the `--my-color` CSS custom property or update its value if it already exists. All other CSS custom properties will remain unchanged. 

You can call `cssVars()` as many times as you like, passing only the custom properties you want to add or update to [`options.variables`](https://github.com/jhildenbiddle/css-vars-ponyfill#optionsvariables). For performance reasons, it's obviously better to call the ponyfill one time with many variables as opposed to multiple times with a single variable each time.

If you want to 1) read a custom property value and then 2) change it based on its value, combine the two methods:

```javascript
// 1st call: Get the CSS custom property values
cssVars({
  onlyLegacy: false,
  updateDOM : false,
  onComplete: function(cssText, styleNode, cssVariables) {
    if (cssVariables["--my-color"] === "red") {
      // 2nd call: Edit a single custom property
      cssVars({
        variables: {
          "--my-color": "blue"
        }
      });
    }
  }
});
```

**Comment by bene-starzengruber** [comment]
Thanks!
I still have the problem that the options.variables is overriding my other variables as well but now with your hints, I just get all variables, change some and set them all again. I know that this is probably not as performant but enough for my use case.

**Comment by jhildenbiddle** [comment]
Fixed in 1.12.1.

I've left the demo on 1.12.0 to you can see the bug. If you update the dependency to 1.12.1, you will see the bug has been fixed.

**Comment by bene-starzengruber** [comment]
I just tried it out with the new version and am still running into the same problem
https://stackblitz.com/edit/angular-u7687e?file=src%2Fapp%2Fapp.component.ts

- Change Background color - Background will change but the css-variable is not set on the :root element

- Change TextColor - Text color is changed but background is resetted to the initial value

**Comment by jhildenbiddle** [comment]
Can you

**Comment by jhildenbiddle** [comment]
Fixed in 1.12.2.

Sorry about that. 1.12.1 fixed a separate but related bug. The bug you ran into was a regression introduced in 1.11.0. This has been fixed, and values passed to `options.variables` should now persist properly.

New demo available here:

* https://stackblitz.com/edit/css-vars-ponyfill-33-2?file=index.js

**Comment by bene-starzengruber** [comment]
Yes, everything is working now!

Thanks, your work is highly appreciated :)
